### ISSUE TITLE
Don't duplicate metadata between light and extended?

### ISSUE THREAD

**Issue by xvrh** [cedvdb/dart_phone_metadata#1]
Hi!
I discover the library and notice one small thing (I might be wrong).
It seems `extendedMetadataByIsoCode` is a super set of `lightMetadataByIsoCode`. And  `extendedMetadata` duplicate all the data from `lightMetadata`.
I'm a bit worried that it is super easy to use both the LightParser and FullParser without noticing. So we will end up pulling both light and extended metadata. Thus, increasing the code size and the memory.

Is it possible to re-organise the metadata so the extendedMetadata only reference the added data. So there is no duplication?
```
const metaData = /* all shared data */
const extendedMetaData = /* only the extra info */
```

**Comment by cedvdb** [comment]
> And extendedMetadata duplicate all the data from lightMetadata.
 
 It's a const so I don't think there is duplicates in memory. [They are canonicalized. This is sort of like string interning: for any given const value, a single const object will be created and re-used no matter how many times the const expression(s) are evaluated.](https://news.dartlang.org/2012/06/const-static-final-oh-my.html)
 
 That means that the shared data between the two are also const and are only once in memory.
 
 If I misunderstood something or if you think the extendedMetadata should still contain only the "extension data", feel free to answer, I'll reopen then.

**Comment by xvrh** [comment]
Thanks for the answer!

I did this super quick experiment:
When I compile this application 
```dart
import 'package:phone_numbers_parser/phone_numbers_parser.dart';

void main() {
  print(PhoneParser().parseWithIsoCode('BE', '477123456'));
  // print(LightPhoneParser().parseWithIsoCode('BE', '477123456')); // Toggle the comment to compare with and without
}
```

`dart compile js -o build/output.js example/treeshaking_test.dart`

I see that the version using both Light and normal PhoneParser is 50Ko greater that the one using only the normal parser.
Since the normal parser directly inherit from the Light one this seems a lot.

When I look at the generated javascript, I can see that the metadata for light and extended is indeed duplicated.

Note: I use the `dart2js` compiler because it is easier to read the output instead of the Flutter/native compilers. But I think dart2js has the same tree-shaking properties than the other compilers.

WDYT?

**Comment by cedvdb** [comment]
Edit here are the results I got trying your snippet right now

```
// uncompressed
Only Phone Parser => 376Kb 
Only Light parser => 172Kb 
Both parsers => 423Kb 
```
And after checking the output I see duplication indeed. I'm not sure this is specific to javascript compilation or not but something needs to be done anyway

**Comment by cedvdb** [comment]
Fixed in future release.

Importing both now results in a js of 244kb (was 423kb before). So a significant decrease, even more than anticipated. Probably the compiler could optimize some things behind the scene.

The light parser is now 149kb (from 172kb). 

Those numbers are from compiling your snippet of course. If you are interested, I splitted the metadatas like this:

https://github.com/cedvdb/dart_phone_metadata/tree/main/lib/src/generated
